scripts/update-projects-version.py

```diff
@@ -0,0 +1,375 @@
+        "https://quay.io/api/v1/repository/pelorus/pelorus-operator/tag/"
+    )
+    pelorus_operator_image_tags.add_header("Authorization", "Bearer XYZ")
+    with request.urlopen(pelorus_operator_image_tags) as response:
+        tags: List[Dict[str, str]] = list(json.load(response)["tags"])
+    tag_names = [
+        tag["name"]
+        for tag in tags
+        if tag.get("end_ts") is None and not tag["name"].upper().isupper()
+    ]
+    tag_names.sort(reverse=True)
+
+    add_replaces_to_csv(
+        new_version=next_operator_version,
+        cluster_service_version_file=destination
+        / "bundle/manifests/pelorus-operator.clusterserviceversion.yaml",
+        versions=tag_names,
+    )
+
+    logging.info(f"Updated operator available at {destination}")
+
+
+def main(arguments: argparse.Namespace) -> None:
+    check_prerequisites(destination=arguments.destination)
+
+    next_chart_version = get_next_version(
+        file="charts/pelorus/Chart.yaml",
+        software="chart",
+        arguments=arguments,
+    )
+    update_charts_folder(
+        next_chart_version=next_chart_version,
+    )
+
+    next_operator_version = get_next_version(
+        file="pelorus-operator/bundle/manifests/pelorus-operator.clusterserviceversion.yaml",
+        software="operator",
+        arguments=arguments,
+    )
+    update_operator_folder(
+        next_operator_version=next_operator_version,
+        destination=arguments.destination,
+    )
+
+    if arguments.major:
+        logging.info('IMPORTANT: label your PR with "major"')
+    if arguments.minor:
+        logging.info('IMPORTANT: label your PR with "minor"')
+    if not arguments.pre_release:
+        documentation_file = ROOT / "docs/Development.md"
+        replace_in_file(
+            file=documentation_file,
+            pattern=VERSION_PATTERN,
+            new=next_chart_version,
+            number_of_changes=2,
+        )
+        logging.info("IMPORTANT: this change will result in a new release")
+    logging.info("Update finished successfully")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Update all versions of the project.",
+        allow_abbrev=False,
+    )
+    parser.add_argument(
+        "-d",
+        "--destination",
+        type=path_type,
+        help="Path to operator destination folder",
+        default=PELORUS_OPERATOR_FOLDER,
+    )
+    parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="Removes destination folder before proceeding.",
+    )
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "-x",
+        "--major",
+        action="store_true",
+        help="Major update to project versions.",
+    )
+    group.add_argument(
+        "-y",
+        "--minor",
+        action="store_true",
+        help="Minor update to project versions.",
+    )
+    group.add_argument(
+        "-z",
```

Comment:
Same here and for the pre-release, we are forcing someone to bump both versions which isn't always necessary. I think it's perfectly valid scenario to keep the helm charts on the next patch release with a new minor release of pelorus-operator.

There was a long discussion about versioning and we never said a major bump of one (helm or operator) will require major bump of another:

https://github.com/dora-metrics/pelorus/discussions/811#discussioncomment-5062792

Reply:
We could release major version of pelorus-operator with minor version bump of helm charts for example. We should never release one with rc version of another.